api/service.capnp

```diff
@@ -0,0 +1,54 @@
+using Go = import "/go.capnp";
+
+@0xfcba4f486a351ac3;
+
+$Go.package("discovery");
+$Go.import("github.com/wetware/ww/internal/api/service");
+
+
+interface ServiceDiscovery {
+    provider @0 (name :Text) -> (provider :Provider);
+    locator @1 (name :Text) -> (locator :Locator);
+}
+
+interface Provider {
+    provide @0 (location :SignedLocation) -> ();
+}
+
+interface Locator {
+    findProviders @0 (chan :Sender(SignedLocation)) -> ();
+    
+    using Sender = import "channel.capnp".Sender;
+}
+
+struct SignedLocation {
+    signature @0 :Data;
+    location  @1 :Location;
+}
+
+struct Location {
+    id @0 :PeerID;
+    union{
+        maddrs @1 :List(Multiaddr);
+        anchor @2 :AnchorPath;
+        custom @3 :AnyPointer;
+    }
+    meta     @4 :List(Text);
```

Comment:
I don't remember why we added this, but I'm getting a whiff of feature-creep (...it was probably my idea lol).

Why don't we remove this until we need it?

Reply:
We want metadata for filtering different responses based on features the users are interested in, such as physical location: But there is much other metadata about service providers that users are interested in.